htdocs/compta/facture/class/api_invoices.class.php

```diff
@@ -632,8 +632,10 @@ public function put($id, $request_data = null)
 			}
 		}
 
-		if ($this->invoice->update(DolibarrApiAccess::$user)) {
+		if ($this->invoice->update(DolibarrApiAccess::$user) > 0) {
 			return $this->get($id);
+		} else {
+			throw new RestException(500, $this->invoice->error);
```

Comment:
If you add a thro, you must remove the return false after, because this code become unreachable.

Reply:
Okay, that makes sense